pkg/client/clientcmd/validation.go

```diff
@@ -109,7 +109,8 @@ func validateClusterInfo(clusterName string, clusterInfo clientcmdapi.Cluster) [
 	if len(clusterInfo.Server) == 0 {
 		validationErrors = append(validationErrors, fmt.Errorf("no server found for %v", clusterName))
 	}
-	if len(clusterInfo.CertificateAuthority) != 0 {
+	// Only validate the CA file if it hasn't been overridden by CA data
```

Comment:
I'd rather see an either or.  I don't think it makes much sense to allow both a CA file and CA data to be specified.